### ISSUE TITLE
📝 [Proposal]: Add Support for the Private Network Access CORS Header `Access-Control-Request-Private-Network`

### ISSUE THREAD

**Issue by GalvinGao** [gofiber/fiber#2804]
### Feature Proposal Description

Add CORS Middleware Support for the Private Network Access CORS Header `Access-Control-Request-Private-Network` to avoid additional preflight/timeout time cost.

### Alignment with Express API

Express.js's [CORS middleware](https://github.com/expressjs/cors) has not been updated yet.

### HTTP RFC Standards Compliance

- [Private Network Access: introducing preflights](https://developer.chrome.com/blog/private-network-access-preflight?hl=en)
- [Private Network Access (W3C Draft Report) (Previously known as CORS-RFC1918)](https://wicg.github.io/private-network-access/)

### API Stability

It seems that Chrome 104 has [already shipped](https://developer.chrome.com/blog/private-network-access-preflight?hl=en#rollout_plan) this header.

### Feature Examples

```go
app.Use(cors.New(cors.Config{
    AllowPrivateNetworkAccess: true,

    // other already existing fields omitted...
}))
```

#### Example Request

```http
OPTIONS /path HTTP/1.1
Accept: */*
Access-Control-Request-Method: GET
Access-Control-Request-Private-Network: true
Origin: https://example.com
Sec-Fetch-Mode: cors
```

#### Expected Response for the Example Request

```http
HTTP/1.1 204 No Content
Access-Control-Allow-Private-Network: true
Vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET,POST,HEAD,PUT,DELETE,PATCH
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have searched for existing issues that describe my proposal before opening this one.
- [X] I understand that a proposal that does not meet these guidelines may be closed without explanation.

**Comment by Behzad-Khokher** [comment]
This is really interesting. But I was just curious how are we avoiding pre-flight requests in this case. To access a private resource wouldn't the browser still need to make an `[options]` pre-flight request with the header `Access-Control-Request-Private-Network: true` for the server would respond with `Access-Control-Allow-Private-Network: true`.

What I'm confused about is that isn't it up to the browser to decide automatically when a pre-flight request needs to be sent. Either way, it would be sending a pre-flight request any way? So how are we avoiding additional pre-flights?

**Comment by the-hotmann** [comment]
@Behzad-Khokher
here the blogpost from Google, on how it works: https://developer.chrome.com/blog/private-network-access-preflight

**Comment by ReneWerner87** [comment]
what would be the specific task of the cors middleware?
to only render this header if this is allowed in the config ?

**Comment by Behzad-Khokher** [comment]
@ReneWerner87 I can look into this. May I self assign?

**Comment by ReneWerner87** [comment]
Thx

**Comment by sixcolors** [comment]
This is still in draft and has not been accepted as a standard. https://wicg.github.io/private-network-access/

**Comment by gaby** [comment]
Fixed via #2908 will be a new festure in Fiber v3
